PYME/config.py

```diff
@@ -279,6 +284,45 @@ def get_plugins(application):
         
     return  list(set([p.strip() for p in plugin_paths if not p.strip() == '']))
 
+def get_report_filters():
+    """
+    Find (jinja2) filters registered as plugins
+
+    Returns
+    -------
+    filter_info: dict
+        keys are modules to import, and values are lists of functions/filters in that module
+    """
+    import yaml
+    filter_info = {}
+    for config_dir in config_dirs:
+        filter_glob = glob.glob(os.path.join(config_dir, 'plugins', 'reports', 'filters/*.yaml'))
+        for plugin_file in filter_glob:
+            with open(plugin_file) as f:
+                filter_info.update(yaml.safe_load(f))
+    return filter_info
+
+def get_report_templates():
+    """
+    Find (jinja2) templates registered as plugins
+
+    Returns
+    -------
+    templates: dict
+        keys are the filename + extension, e.g. template.html, and values are the full path
+    """
+    templates = {}
+    for config_dir in config_dirs:
+        template_glob = glob.glob(os.path.join(config_dir, 'plugins', 'reports', 'templates/*.txt'))
+
+        for fn in template_glob:
+            with open(fn, 'r') as f:
+                for directory_listed in f:
+                    found = glob.glob(os.path.join(directory_listed, '*.html'))
```

Comment:
I'd prefer not to force templates to be html. There's nothing about jinja which forces things to be html, and adding, e.g. a PDF output using templated latex and latex2pdf is on the wishlist (reasonably far down, but I've done similar things in the past and don't want our design to preclude this). Even if we stick to HTML, there are at least 3 valid file extensions (.htm, .html, .xhtml).

Reply:
Probably the best way of solving this is to just have `UnifiedLoader` try all the directories we have listed as options. 

The other option is to stick with the status quo and require an explicit path to templates not stored in `recipes/templates`.

Reply:
In either case, looking up from a global dictionary which is keyed by template name (with no path) is likely to problematic, as the likelihood of name collisions will be high. The chance that two plugins will elect to both call their template "report.html" is much greater than 0.

I'd suggest that plugin templates should need to be accessed through the module namespace, e.g. `nep_fitting/report.html`.

Reply:
Things are now setup (and documented) to access templates through the module namespace with no limitations on extensions.